src/optimalcontrol/tests/OptimizerTest.cpp

```diff
@@ -0,0 +1,203 @@
+        costValue = m_a1 * m_variables(0) * m_variables(0) + m_b1 * m_variables(0) + m_a2 * m_variables(1) * m_variables(1) + m_b2 * m_variables(1);
+        return true;
+    }
+
+    virtual bool evaluateCostGradient(iDynTree::VectorDynSize& gradient) override {
+        gradient.resize(numberOfVariables());
+        gradient(0) = 2 * m_a1 * m_variables(0) + m_b1;
+        gradient(1) = 2 * m_a2 * m_variables(1) + m_b2;
+        return true;
+    }
+
+    virtual bool evaluateCostHessian(iDynTree::MatrixDynSize& hessian) override {
+        hessian.resize(2,2);
+        hessian.zero();
+        hessian(0,0) = 2*m_a1;
+        hessian(1,1) = 2*m_a1;
+        return true;
+    }
+
+    virtual bool evaluateConstraints(iDynTree::VectorDynSize& constraints) override {
+        constraints.resize(2);
+        constraints = m_variables;
+        return true;
+    }
+
+    virtual bool evaluateConstraintsJacobian(iDynTree::MatrixDynSize& jacobian) override {
+        jacobian.resize(2,2);
+        jacobian(0,0) = 1.0;
+        jacobian(1,1) = 1.0;
+        return true;
+    } //using dense matrices, but the sparsity pattern is still obtained
+
+    virtual bool evaluateConstraintsHessian(const iDynTree::VectorDynSize& constraintsMultipliers, iDynTree::MatrixDynSize& hessian) override {
+        hessian.resize(2,2);
+        hessian.zero();
+        return true;
+    } //using dense matrices, but the sparsity pattern is still obtained
+
+    bool setPlusInfinity(double plusInfinity) {
+        if (plusInfinity < 0)
+            return false;
+
+        m_plusInfinity = plusInfinity;
+        return true;
+    }
+
+    bool setMinusInfinity(double minusInfinity) {
+        if (minusInfinity > 0)
+            return false;
+
+        m_minusInfinity = minusInfinity;
+        return true;
+    }
+
+    double expectedMinimum() {
+        double expected;
+        if (m_c1 < (-m_b1/(2*m_a1)))
+            expected = - m_b1 * m_b1 / (4*m_a1);
+        else
+            expected = m_a1*m_c1*m_c1 + m_b1*m_c1;
+
+        if (m_c2 < (-m_b2/(2*m_a2)))
+            expected += - m_b2 * m_b2 / (4*m_a2);
+        else
+            expected += m_a2*m_c2*m_c2 + m_b2*m_c2;
+
+        return expected;
+    }
+
+    const iDynTree::VectorDynSize& expectedVariables() {
+        m_expectedVariables.resize(2);
+        if (m_c1 < (-m_b1/(2*m_a1)))
+            m_expectedVariables(0) = - m_b1 / (2*m_a1);
+        else
+            m_expectedVariables(0) = m_c1;
+
+        if (m_c2 < (-m_b2/(2*m_a2)))
+            m_expectedVariables(1) = - m_b2 / (2*m_a2);
+        else
+            m_expectedVariables(1) = m_c2;
+
+        return m_expectedVariables;
+    }
+};
+
+int main(){
+    iDynTree::optimization::IpoptInterface ipoptSolver;
+    std::shared_ptr<TestProblem> problem(new TestProblem);
+    iDynTree::VectorDynSize guess(2), dummy1, dummy2, dummy3;
+    guess.zero();
+
+    iDynTree::assertTrue(ipoptSolver.setIpoptOption("nlp_lower_bound_inf", -1.0e20));
```

Comment:
If you use the `ASSERT_TRUE` macros, in case of error you also get the line in which the failed assert occured.